src/components/button/braintree.js

```diff
@@ -0,0 +1,34 @@
+/* @flow */
+
+import { SyncPromise } from 'sync-browser-mocks/src/promise';
+
+export type Braintree = {
+    client : {
+        create : Function
+    },
+    paypalCheckout : {
+        create : Function
+    }
+};
+
+export type BraintreePayPalClient = {
+    createPayment : Function,
+    tokenizePayment : Function
+};
+
+export function awaitBraintreeClient(braintree : Braintree, authorization : string) : SyncPromise<BraintreePayPalClient> {
+    return new SyncPromise((resolve, reject) => {
+
+        braintree.client.create({ authorization }, (err, client) => {
```

Comment:
OK changed to use promises instead. Going to merge this soon if no further thoughts. I've added some demo docs to our staging env at http://devdocs-4307.ccg21.dev.paypalcorp.com:8888/docs/integration/direct/express-checkout/integration-jsv4/braintree-sdk/